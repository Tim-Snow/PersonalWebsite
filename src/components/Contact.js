import React from 'react'

import Button from '@material-ui/core/Button'

function Contact() {
  return (
    <div style={styles.container}>
      <h2>Contact me</h2>
      <form style={styles.form}>
        <input
          style={styles.margin}
          placeholder="Your name"
          type="text"
          name="name"
        />
        <input
          style={styles.margin}
          type="text"
          placeholder="Subject"
          name="subject"
        />
        <textarea
          style={styles.margin}
          placeholder="Your message"
          rows="8"
          cols="80"
          name="message"
        />

        <Button variant="contained" color="primary" style={styles.margin}>
          Send
        </Button>
      </form>
    </div>
  )
}

const styles = {
  container: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    padding: 15,
    backgroundColor: '#DFDFDF',
    marginTop: 10,
    marginLeft: 10,
    marginRight: 10,
  },
  form: {
    display: 'flex',
    flexDirection: 'column',
  },
  margin: {
    margin: 5,
  },
}

export default Contact
