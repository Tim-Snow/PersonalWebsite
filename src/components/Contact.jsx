import React, { useState } from 'react'

import Button from '@material-ui/core/Button'
import { radius, secondary, shadow } from '../constants/styles'

function Contact() {
  const [sent, setSent] = useState(false)

  function handleSubmit() {
    setSent(true)
  }

  return (
    <div style={styles.container}>
      <h2>Contact me</h2>
      <div style={styles.form}>
        <input
          style={styles.margin}
          placeholder="Your name"
          type="text"
          name="name"
        />
        <input
          style={styles.margin}
          type="text"
          placeholder="Subject"
          name="subject"
        />
        <textarea
          style={styles.margin}
          placeholder="Your message"
          rows="8"
          cols="80"
          name="message"
        />
      </div>
      {sent && (
        <div style={styles.error}>
          <p>
            Sorry, I haven't implemented anything but the visuals for this,
            yet..!
          </p>
          <p>
            Use the info at the top of the page to get in contact{' '}
            <span role="img" aria-label="smile">
              ðŸ˜Š
            </span>
          </p>
        </div>
      )}
      <Button
        variant="contained"
        color="primary"
        style={styles.margin}
        onClick={handleSubmit}
      >
        Send
      </Button>
    </div>
  )
}

const styles = {
  container: {
    borderRadius: radius,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    padding: 15,
    backgroundColor: secondary,
    marginTop: 10,
    marginLeft: 10,
    marginRight: 10,
    boxShadow: shadow,
  },
  form: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  margin: {
    margin: 5,
  },
  error: {
    textAlign: 'center',
    color: '#F33',
  },
}

export default Contact
