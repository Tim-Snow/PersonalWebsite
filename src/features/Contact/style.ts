import { secondary } from 'constants/styles';

const style = {
  container: {
    borderRadius: 10,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    padding: 15,
    backgroundColor: secondary,
    marginTop: 10,
    marginLeft: 10,
    marginRight: 10,
  },
  form: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  margin: {
    margin: 5,
  },
  error: {
    textAlign: 'center',
    color: '#F33',
  },
} as { [key: string]: React.CSSProperties };

export default style;
