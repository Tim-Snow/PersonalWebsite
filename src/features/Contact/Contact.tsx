import React, { useState } from 'react';
import Button from '@material-ui/core/Button';

import style from './style';

export default function Contact() {
  const [sent, setSent] = useState(false);

  const handleSubmit = () => {
    setSent(true);
  };

  return (
    <div style={style.container}>
      <h2>Contact me</h2>
      <div style={style.form}>
        <input style={style.margin} placeholder="Your name" type="text" name="name" />
        <input style={style.margin} placeholder="Your email" type="text" name="email" />
        <input style={style.margin} type="text" placeholder="Subject" name="subject" />
        <textarea style={style.margin} placeholder="Your message" rows={8} cols={80} name="message" />
      </div>

      {sent && (
        <div style={style.error}>
          <p>Sorry, I haven&apos;t implemented anything but the visuals for this, yet..!</p>
          <p>
            Use the info at the top of the page to get in contact{' '}
            <span role="img" aria-label="smile">
              ðŸ˜Š
            </span>
          </p>
        </div>
      )}

      <Button variant="contained" color="primary" style={style.margin} onClick={handleSubmit}>
        Send
      </Button>
    </div>
  );
}
