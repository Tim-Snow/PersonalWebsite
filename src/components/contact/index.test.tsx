import * as React from 'react'
import * as Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'

import Contact from './index'

configure({ adapter: new Adapter() })

describe('contact', () => {
  it('should match the snapshot', () => {
    const headerlink = shallow(<Contact />)
    expect(headerlink).toMatchSnapshot()
  })
})
