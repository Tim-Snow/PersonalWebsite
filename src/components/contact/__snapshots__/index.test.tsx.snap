// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Contact />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Name",
        <input
          type="name"
        />,
        <br />,
        "Email",
        <input
          type="email"
        />,
        <br />,
        "Phone",
        <input
          type="phone"
        />,
        <br />,
        "Message",
        <textarea />,
        <br />,
        <input
          type="submit"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      "Name",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "type": "name",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      "Email",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "type": "email",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      "Phone",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "type": "phone",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      "Message",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "type": "submit",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Name",
          <input
            type="name"
          />,
          <br />,
          "Email",
          <input
            type="email"
          />,
          <br />,
          "Phone",
          <input
            type="phone"
          />,
          <br />,
          "Message",
          <textarea />,
          <br />,
          <input
            type="submit"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        "Name",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "type": "name",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        "Email",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "type": "email",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        "Phone",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "type": "phone",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        "Message",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "type": "submit",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
